### DISCUSSION TITLE
Remapping buttons for stylus

### DISCUSSION THREAD

**Discussion by onefl0w** [flxzt/rnote#127] [Q&A]
Hello !

Great app, everything works.

I'd like to know how could I remap the behavior of my HP Active Pen buttons ?

Is it possible that when I hold on the first button it erase and when I hold the second it select ?

How can I implement that ?

Thanks ;)

**Comment by flxzt** [comment]
configurable stylus button actions are now included in v0.4.0 :) Look at the settings page to change it to your liking.

**Reply by pj-camp** [reply]
Except if the thing you like is to completely disable the buttons. I stopped using Openboard because of this. It literally ignores the OS configuration for the buttons and I'm tired of the eraser popping up while drawing. Xournal++ allows me to set both stylus buttons to do nothing at all. But Xournal++ is page oriented and you can't change that. Infinite canvas is not an option. That's why I decided to look at Rnote, but I can't use it if I can't disable the stylus buttons.

**Reply by Doublonmousse** [reply]
You should be able to disable buttons ? (last option in the settings).

![image](https://github.com/user-attachments/assets/43f3b6b4-334c-4b30-9acd-9b75ba96e523)

The only thing that may not be disabled is the pen/eraser distinction (hence a button that is mapped to an eraser instead of a button will not be disabled as from the viewpoint of note it won't be seen as a button press). IMHO it shouldn't happen in the first place. I guess that's your issue, right?

Edit : There's a nuance to that, from https://github.com/flxzt/rnote/issues/1104 (and the related yet-to-be merged PR, you can force the eraser to always act as a pen, but I'm not sure this would actually fix your issue, not sure you'd be able to write a single continuous path if the eraser is wrongly activated/deactivated at some point during this)

**Comment by pj-camp** [comment]
Those options are not available on Openboard. For some reason, setting the
button 1 action to 0 in Linux causes Openboard to stop recognizing the
stylus as something that can be used for drawing but the button still
brings up the eraser.

They are available on Xournal++ which I started test driving this week. I
think that's where you got the screenshot. So far I like it except for not
having an infinite canvas.

On Tue, 24 Sept 2024 at 00:11, Doublonmousse ***@***.***>
wrote:

> You should be able to disable buttons ? (last option in the settings).
>
> image.png (view on web)
> <https://github.com/user-attachments/assets/43f3b6b4-334c-4b30-9acd-9b75ba96e523>
>
> The only thing that may not be disabled is the pen/eraser distinction
> (hence a button that is mapped to an eraser instead of a button will not be
> disabled as from the viewpoint of note it won't be seen as a button press).
> IMHO it shouldn't happen in the first place. I guess that's your issue,
> right?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/flxzt/rnote/discussions/127#discussioncomment-10733784>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANOYDJXUWW52Y5AYWXOUPCLZYDQ55AVCNFSM6AAAAABOWWVNBGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANZTGM3TQNA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Dr. Paul J. Camp

"The beauty of the cosmos derives not only from unity in variety
       but also from variety in unity"
                 -- Umberto Eco
                     The Name of the Rose

**Reply by Doublonmousse** [reply]
This is a note screenshot ! 
![image](https://github.com/user-attachments/assets/bcc982f7-f731-4e4e-b76f-f01ce099d15b)
I had to recheck but it's available as an option in 0.11

**Comment by pj-camp** [comment]
Oh I see! I'm getting my various stylus threads confused. I'll take a new
look at Rnote. I guess I haven't looked at it for a while. Thanks!

On Tue, 24 Sept 2024 at 12:55, Doublonmousse ***@***.***>
wrote:

> This is a note screenshot !
> image.png (view on web)
> <https://github.com/user-attachments/assets/bcc982f7-f731-4e4e-b76f-f01ce099d15b>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/flxzt/rnote/discussions/127#discussioncomment-10741932>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANOYDJQH6FWO5UQJSKVUA6LZYGKPXAVCNFSM6AAAAABOWWVNBGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANZUGE4TGMQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Dr. Paul J. Camp

"The beauty of the cosmos derives not only from unity in variety
       but also from variety in unity"
                 -- Umberto Eco
                     The Name of the Rose
